cmd/openfga/openfga_test.go

```diff
@@ -64,11 +63,12 @@ func TestBuildServerWithPresharedKeyAuthentication(t *testing.T) {
 	require.NoError(t, err, "Failed to build server and/or datastore")
 	defer service.Close(ctx)
 
-	g, ctx := errgroup.WithContext(ctx)
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
This fixes a bug that existed before. `errgroup.WithContext(...)` only cancels the context when the first non-nil error is returned from the Go call or when `g.Wait()` returns. None of these conditions apply, because the context never gets cancelled and thus the server never stops running, and so the Go function never returns.

This changes this behavior and fixes a bug so that tests don't accidentally overlap one another (since they run on the same port).